src/librustc_resolve/lib.rs

```diff
@@ -1666,13 +1666,22 @@ impl<'a, 'crateloader: 'a> Resolver<'a, 'crateloader> {
         let mut extern_prelude: FxHashSet<Name> =
             session.opts.externs.iter().map(|kv| Symbol::intern(kv.0)).collect();
 
-        // HACK(eddyb) this ignore the `no_{core,std}` attributes.
-        // FIXME(eddyb) warn (elsewhere) if core/std is used with `no_{core,std}`.
-        // if !attr::contains_name(&krate.attrs, "no_core") {
-        // if !attr::contains_name(&krate.attrs, "no_std") {
-        extern_prelude.insert(Symbol::intern("core"));
-        extern_prelude.insert(Symbol::intern("std"));
-        extern_prelude.insert(Symbol::intern("meta"));
+        if session.rust_2018() {
+            // HACK(eddyb) this ignore the `no_{core,std}` attributes.
+            // FIXME(eddyb) warn (elsewhere) if core/std is used with `no_{core,std}`.
+            extern_prelude.insert(Symbol::intern("core"));
+            extern_prelude.insert(Symbol::intern("std"));
+            extern_prelude.insert(Symbol::intern("meta"));
+        } else {
+            if !attr::contains_name(&krate.attrs, "no_core") {
+                if !attr::contains_name(&krate.attrs, "no_std") {
+                    extern_prelude.insert(Symbol::intern("std"));
+                    extern_prelude.insert(Symbol::intern("core"));
+                } else {
```

Comment:
This seems like a weird way to write it?

```rust
                extern_prelude.insert(Symbol::intern("core"));

                if !attr::contains_name(&krate.attrs, "no_std") {
                    extern_prelude.insert(Symbol::intern("std"));
                }
```

Reply:
Oh. Well, yes.
I've just reverted to the old version of code an it had that.
Will fix.